### PR TITLE
Sch year 1 clinical data

### PR DESCRIPTION
**Pull request by kairstenfay**
This relies on changes in seattleflu/id3c/pull/88.

Move the clinical, etl clinical, longitudinal, and etl longitudinal commands (and related helper functions) from ID3C to ID3C-customizations.

Make necessary changes to `clinical` and `etl clinical` to ingest new SCH Year 1 data.

### CODE REVIEW

File: lib/seattleflu/id3c/cli/command/__init__.py [hunk 1]
```diff
@@ -7,6 +7,113 @@
...
+import pandas as pd
+from typing import List
+
+# Load all ETL subcommands.
+__all__ = [
+    "clinical",
+    "longitudinal",
+    "reportable_conditions"
+]
+
```
Comment (kairstenfay) [inline lib/seattleflu/id3c/cli/command/__init__.py]: Here, I went with the "old style" of `__init__` subcommand loading because
1. I don't fully understand how the "new style" works, and 
2. I had a boatload of other helper functions to import.

Thoughts? Suggestions?

File: lib/seattleflu/id3c/cli/command/clinical.py [hunk 2]
```diff
@@ -301,6 +299,14 @@ def drop_unnecessary_columns(df: pd.DataFrame, columns_to_keep = List[str]) -> p
     return df[columns_to_keep]
 
 
+def insert_static_value_columns(df: pd.DataFrame, column_values: Dict[str, Any]) -> pd.DataFrame:
+    """ TODO """
```
Comment (kairstenfay) [inline lib/seattleflu/id3c/cli/command/clinical.py]: I'll fix this TODO

File: lib/seattleflu/id3c/cli/command/clinical.py [hunk 3]
```diff
@@ -264,31 +265,47 @@ def parse_sch(sch_filename, output):
...
-
     clinical_records = insert_static_value_columns(clinical_records, {'site': 'SCH'})
     clinical_records = create_encounter_identifier(clinical_records)
     clinical_records = remove_pii(clinical_records)
 
     dump_ndjson(clinical_records)
 
 
+def add_insurance(df: pd.DataFrame) -> pd.DataFrame:
+    """ """ # TODO
```
Comment (kairstenfay) [inline lib/seattleflu/id3c/cli/command/clinical.py]: will handle this TODO

File: lib/seattleflu/id3c/cli/command/clinical.py [hunk 4]
```diff
@@ -264,31 +265,47 @@ def parse_sch(sch_filename, output):
...
+    df['MedicalInsurance'] = df.apply(insurance, axis=1)
+    return df
+
+
+def insurance(series: pd.Series) -> pd.Series:
+    """ """ # TODO
```
Comment (kairstenfay) [inline lib/seattleflu/id3c/cli/command/clinical.py]: will handle this TODO

**Comment by kairstenfay** [comment]
[Ingest new SCH clinical data](https://trello.com/c/nAUlByxY/266-ingest-new-sch-clinical-data)

**Comment by tsibley** [comment]
Rebased this onto master to pick up new changes and brought in new changes from https://github.com/seattleflu/id3c/pull/88

File: lib/seattleflu/id3c/cli/command/clinical.py [hunk 5]
```diff
@@ -291,6 +292,15 @@ def parse_sch(sch_filename, output):
 
 
 
+
+def drop_unnecessary_columns(df: pd.DataFrame, columns_to_keep = List[str]) -> pd.DataFrame:
+    """
+    Drops all columns from a given *df* except those explicitly listed in *columns_to_keep*. Returns
+    the new DataFrame.
+    """
+    return df[columns_to_keep]
```
Comment (tsibley) [inline lib/seattleflu/id3c/cli/command/clinical.py]: On the face of it, this seems like refactoring gone too far. What's the motivation here? (The commit message doesn't say why, just what. :-)

File: lib/seattleflu/id3c/cli/command/clinical.py [hunk 6]
```diff
@@ -359,9 +368,7 @@ def parse_kp(kp_filename, kp_specimen_manifest_filename, output):
     # Convert dtypes
     clinical_records["encountered"] = pd.to_datetime(clinical_records["encountered"])
 
-    # Insert static value columns
-    clinical_records["site"] = "KP"
-
+    clinical_records = insert_static_value_columns(clinical_records, {"site": "KP"})
```
Comment (tsibley) [inline lib/seattleflu/id3c/cli/command/clinical.py]: Ditto with this refactoring. It's not clear to me why this change is an improvement.

File: lib/seattleflu/id3c/cli/command/clinical.py [hunk 7]
```diff
@@ -264,31 +265,49 @@ def parse_sch(sch_filename, output):
...
+    Adds a new column for insurance type to a given *df*. Returns the new
+    DataFrame.
+    """
+    def insurance(series: pd.Series) -> pd.Series:
+        """ Returns an array of unique insurance types from a given *series*. """
+        insurance_columns = ['insurance_1', 'insurance_2']
+        insurances = [ series[i] for i in insurance_columns if not pd.isna(series[i])]
+        return list(set(insurances))
+
+    df['MedicalInsurance'] = df.apply(insurance, axis=1)
```
Comment (tsibley) [inline lib/seattleflu/id3c/cli/command/clinical.py]: Using `axis='columns'` is clearer than `axis=1`, at least to me. :-)

File: lib/seattleflu/id3c/cli/command/clinical.py (hunk 5 above)
Comment (kairstenfay) [reply lib/seattleflu/id3c/cli/command/clinical.py]: That's fair. I thought it made the code more readable. By refactoring it and naming the function after what it does, I removed a comment that said "Drop unnecessary columns".
If you think that's maybe overkill then I understand.

File: lib/seattleflu/id3c/cli/command/clinical.py (hunk 6 above)
Comment (kairstenfay) [reply lib/seattleflu/id3c/cli/command/clinical.py]: Yeah.. this made more sense when we were adding multiple static values to a DataFrame across multiple lines, but the motivation is no longer clear to me for this refactor. I'm ok with dropping this commit.

File: lib/seattleflu/id3c/cli/command/clinical.py (hunk 5 above)
Comment (tsibley) [reply lib/seattleflu/id3c/cli/command/clinical.py]: Nod. I think it's skewing a little too far. Indexing/subsetting operations like `df[columns]` are a pretty fundamental syntax. Wrapping that up into a function means more cognitive overhead (for me) because I have to recall if the function is just `df[columns]` or something more involved (with the latter being my assumed default since it's a function). It also expands a fundamental op from 2 lines (comment + code) to 7 (function def + call).

File: lib/seattleflu/id3c/cli/command/__init__.py (hunk 1 above)
Comment (tsibley) [reply lib/seattleflu/id3c/cli/command/__init__.py]: I'm not sure what you mean by new vs. old style…? I think the code looks fine, though.
